modules/Movecraft/src/main/java/net/countercraft/movecraft/sign/RepairSign.java

```diff
@@ -1,6 +1,265 @@
+import net.countercraft.movecraft.localisation.I18nSupport;
+import net.countercraft.movecraft.mapUpdater.update.UpdateCommand;
+import net.countercraft.movecraft.mapUpdater.update.WorldEditUpdateCommand;
+import net.countercraft.movecraft.repair.Repair;
+import net.countercraft.movecraft.repair.RepairManager;
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.bukkit.ChatColor;
+import org.bukkit.Material;
+import org.bukkit.World;
+import org.bukkit.block.Block;
+import org.bukkit.block.BlockState;
+import org.bukkit.block.Sign;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
 import org.bukkit.event.Listener;
+import org.bukkit.event.block.Action;
+import org.bukkit.event.block.SignChangeEvent;
+import org.bukkit.event.player.PlayerInteractEvent;
+import org.bukkit.inventory.InventoryHolder;
+import org.bukkit.inventory.ItemStack;
+
+import java.util.*;
 
 public class RepairSign implements Listener{
+    private String HEADER = "Repair:";
+    private HashMap<UUID, Long> playerInteractTimeMap = new HashMap<>();//Players must be assigned by the UUID, or NullPointerExceptions are thrown
+
+    @EventHandler
+    public void onSignChange(SignChangeEvent event){
+        if (!event.getLine(0).equalsIgnoreCase(HEADER)){
+            return;
+        }
+        if (event.getLine(1).isEmpty()){
+            event.getPlayer().sendMessage("You must specify a repair state name on second line");
+            event.setCancelled(true);
+            return;
+        }
+    }
+
+    @EventHandler
+    public void onSignClick(PlayerInteractEvent event){
+        String firstLine = "";
+        if (event.getAction() == Action.RIGHT_CLICK_BLOCK || event.getAction() == Action.LEFT_CLICK_BLOCK) {
+            BlockState state = event.getClickedBlock().getState();
+            if (state instanceof Sign) {
+                Sign sign = (Sign) event.getClickedBlock().getState();
+                String signText = ChatColor.stripColor(sign.getLine(0));
+
+                if (signText == null) {
+                    return;
+                }
+            }
+        }
+        if (event.getAction() == Action.RIGHT_CLICK_BLOCK) {
+            BlockState state = event.getClickedBlock().getState();
+            if (state instanceof Sign) {
+                signRightClick(event);
+            }
+            return;
+        }
+        if (event.getAction() == Action.LEFT_CLICK_BLOCK) {
+            BlockState state = event.getClickedBlock().getState();
+            if (state instanceof Sign) {
+                signLeftClick(event);
+            }
+        }
+    }
+    //
+    private void signLeftClick(PlayerInteractEvent event){
+
+        Sign sign = (Sign) event.getClickedBlock().getState();
+        if (!sign.getLine(0).equalsIgnoreCase(HEADER) || sign.getLine(0) == null){
+            return;
+        }
+        String repairName = event.getPlayer().getName();
+        repairName += "_";
```

Comment:
Might make sense to create a function to get or save the repair schematic from a sign since it is used in both save and load (simplifies changes to either and reduces the chance of bugs in one or the other).

Reply:
This also could allow you to parse the file name there and avoid issues with dangerous characters